Resources/Prototypes/_Impstation/Objectives/thief.yml

```diff
@@ -0,0 +1,453 @@
+- type: entity
+  parent: BaseThiefStealAnimalObjective
+  id: FinFinStealObjective
+  components:
+  - type: StealCondition
+    stealGroup: AnimalFinFin
+  - type: Objective
+    difficulty: 2
+
+- type: entity
+  parent: BaseThiefStealStructureObjective
+  id: GeneratorStealObjective #tesla or singulo generator
+  components:
+  - type: NotJobRequirement
+    job: StationEngineer
+  - type: StealCondition
+    stealGroup: Generator
+    descriptionText: objective-condition-thief-generator-description
+  - type: Objective
+    difficulty: 1.5
+
+- type: entity
+  parent: BaseThiefStealObjective
+  id: PunpunIDCardStealObjective
+  components:
+  - type: NotJobRequirement
+    job: Bartender
+  - type: StealCondition
+    stealGroup: PunpunID
+  - type: Objective
+    difficulty: 0.1
+
+- type: entity
+  parent: BaseThiefStealCollectionObjective
+  id: CigaretteStealCollectionObjective
+  components:
+  - type: SpeciesRequirement
+    allowedSpecies:
+      - Gray
+  - type: StealCondition
+    stealGroup: Cigarettes
+    descriptionText: objective-condition-thief-cigarettes
+    minCollectionSize: 100
+    maxCollectionSize: 1000
+    verifyMapExistence: false
+  - type: Objective
+    difficulty: 0.3
+
+- type: entity
+  parent: BaseThiefStealCollectionObjective
+  id: PositronicBrainStealCollectionObjective
+  components:
+  - type: SpeciesRequirement
+    allowedSpecies:
+      - SpElf
+  - type: NotJobRequirement
+    job: Scientist
+  - type: StealCondition
+    stealGroup: PositronicBrains
+    minCollectionSize: 1
+    maxCollectionSize: 3
+    verifyMapExistence: false
+  - type: Objective
+    difficulty: 2
+
+- type: entity
+  parent: BaseThiefStealCollectionObjective
+  id: AILawBoardStealCollectionObjective
+  components:
+  - type: NotJobRequirement
+    job: Scientist
+  - type: StealCondition
+    stealGroup: AILawBoard
+    minCollectionSize: 1
+    maxCollectionSize: 7
+  - type: Objective
+    difficulty: 1.6
+
+- type: entity
+  parent: BaseThiefStealStructureObjective
+  id: AICoreStealObjective
+  components:
+  - type: NotJobRequirement
+    job: Scientist
+  - type: StealCondition
+    stealGroup: AICore
+  - type: Objective
+    difficulty: 2.5 #im making this high difficulty so its rare even though objectively it's pretty easy
+
+- type: entity
+  parent: BaseThiefStealStructureObjective
+  id: ParticleAcceleratorStealObjective #lol
+  components:
+  - type: NotJobRequirement
+    job: StationEngineer
+  - type: StealCondition
+    stealGroup: ParticleAccelerator
+    descriptionText: objective-condition-thief-multiply-structure-description
+    minCollectionSize: 1
+    maxCollectionSize: 5
+    verifyMapExistence: false
+  - type: Objective
+    difficulty: 3 #good luck dude
```

Comment:
thief objectives are capped at 2.5 combined difficulty: this cannot roll

Reply:
wack, there's a commented out traitor objective that was 5 difficulty so i was operating on that knowledge. explains why i never rolled this in testing